### PR TITLE
[#29] Implement dependency injection mechanism

### PR DESCRIPTION
**Pull request by Sheptytskyid**

### CODE REVIEW

File: src/main/java/com/manardenza/utils/Injector.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+public class Injector {
+    private static File hotelsDatabaseFile = new File("src/main/resources/databaseFiles/databaseHotelsFile.bin");
+    private static File usersDatabaseFile = new File("src/main/resources/databaseFiles/databaseUsersFile.bin");
+    @Getter
+    private static HotelDaoImpl hotelDao = new HotelDaoImpl(hotelsDatabaseFile);
+    @Getter
+    private static UserDaoImpl userDao = new UserDaoImpl(usersDatabaseFile);
+    private static HotelService hotelService = new HotelService(hotelDao);
+    private static UserService userService = new UserService(userDao, CurrentUser.getInstance());
+    private static HotelController hotelController = new HotelController(hotelService);
```
Comment (Sheptytskyid) [inline src/main/java/com/manardenza/utils/Injector.java:23]: MAJOR Remove this unused "hotelController" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/com/manardenza/utils/Injector.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    private static UserController userController = new UserController(userService);
```
Comment (Sheptytskyid) [inline src/main/java/com/manardenza/utils/Injector.java:24]: MAJOR Remove this unused "userController" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/com/manardenza/utils/Injector.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import com.manardenza.dao.HotelDaoImpl;
+import com.manardenza.dao.UserDaoImpl;
+import com.manardenza.login.CurrentUser;
+import com.manardenza.service.HotelService;
+import com.manardenza.service.UserService;
+import lombok.Getter;
+
+import java.io.File;
+
+public class Injector {
```
Comment (Sheptytskyid) [inline src/main/java/com/manardenza/utils/Injector.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1118)

File: src/main/java/com/manardenza/Main.java [hunk 4]
```diff
@@ -8,6 +9,7 @@
 
     public static void main(String[] args) {
         System.out.println("Hello world!");
+        Injector injector = new Injector();
```
Comment (Sheptytskyid) [inline src/main/java/com/manardenza/Main.java]: MAJOR Remove this useless assignment to local variable "injector". [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1854)
MINOR Remove this unused "injector" local variable. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1481)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/manardenza/Main.java (hunk 4 above)
Comment (zozich) [inline src/main/java/com/manardenza/Main.java]: Considering that all variables inside are static, you don't even need it's instance

**Review by zozich — COMMENTED** [review]
Please check

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Review by zozich — APPROVED** [review]
Ok

**Pull request merged**